elements/pfelement/src/pfelement.js

```diff
@@ -1,11 +1,31 @@
 import { autoReveal } from "./reveal.js";
-const prefix = "pfe-";
+import { isAllowedType, isValidDefaultType } from "./attrDefValidators.js";
+
+const tagPrefix = "pfe-";
 
 class PFElement extends HTMLElement {
   static create(pfe) {
+    pfe._validateProperties();
     window.customElements.define(pfe.tag, pfe);
   }
 
+  /**
+   * This prefix is prepended to the name of any attributes defined on the base
+   * class.  In other words, attributes that can be used on any element will be
+   * prefixed with this string.
+   *
+   * @example A "theme" prop, when reflected as an attribute, prefixed with "pfe-g-": <pfe-foo pfe-g-theme="bar">
+   */
+  static get globalAttrPrefix() {
+    return "pfe-g-";
+  }
+
+  static get attrPrefix() {
+    return "pfe-c-";
```

Comment:
```suggestion
    return `${prefix}-c-`;
```

Reply:
I renamed `prefix` to `tagPrefix` to differentiate it from the attribute prefixes, but you're right that if we ever were to change the `pfe-` prefix on tags, it would also make sense to change the prefixes on attributes.  That seems like a good reason to change the name back to `prefix`.  Or, more accurately, prefixPrefix, since it's prefixing the attribute prefix. :laughing: